File: java/src/test/java/testingil/webinar/tddapi/s08/added_test/NavigatorUnitTests.java
package testingil.webinar.tddapi.s08.added_test;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import testingil.webinar.tddapi.s08.Distance;
import testingil.webinar.tddapi.s08.Location;
import testingil.webinar.tddapi.s08.Navigator;

import static org.hamcrest.MatcherAssert.assertThat;

class NavigatorUnitTests {

	@Test
	void when_created_distance_is_always_zero() {
		Location loc = new Location("London");
		Navigator nav = new Navigator(loc);
		Distance distance = nav.getDistanceFromDestination();
		assertThat(distance.inKm(), is(0));
	}
	
	@Test
	@DisplayName("2500 miles = 4000 kms")
	void distance_is_calculated_in_km() {
		Location startingPoint = new Location("New York City");
		Navigator nav = new Navigator(startingPoint);
		
		Location dest = new Location("Los Angeles");
		nav.setDestination(dest);
		
		Distance distance = nav.getDistanceFromDestination();
		assertThat(distance.inKm(), is(4000));
	}

}

File: Python/s07_ut_passing/navigator_unit_tests.py
import unittest

from hamcrest import assert_that, raises, calling, equal_to

from s07_ut_passing.location import Location
from s07_ut_passing.navigator import Navigator


class NavigatorUnitTests(unittest.TestCase):
    def test_when_created_distance_is_always_zero(self):
        loc = Location("London")
        nav = Navigator(loc)
        distance = nav.getDistanceFromDestination()
        assert_that(distance.inKm(), equal_to(0))


if __name__ == '__main__':
    unittest.main()
